server_fn_macro/src/lib.rs

```diff
@@ -308,15 +306,15 @@ pub fn server_macro_impl(
         }
     };
 
-    // TODO rkyv derives
-    let derives = if input_is_multipart {
-        quote! {}
-    } else if input_is_rkyv {
-        todo!("implement derives for Rkyv")
-    } else {
-        quote! {
+    let derives = match input_ident.to_string().as_str() {
+        "Rkyv" => todo!("implement derives for Rkyv"),
+        "MultipartFormData" => quote! {},
+        "SerdeLite" => quote! {
+            Clone, #server_fn_path::serde_lite::Serialize, #server_fn_path::serde_lite::Deserialize
+        },
+        _ => quote! {
             Clone, #server_fn_path::serde::Serialize, #server_fn_path::serde::Deserialize
-        }
+        },
     };
     let serde_path = (!input_is_multipart && !input_is_rkyv).then(|| {
```

Comment:
Okay, what I've just pushed won't compile. I wanted to understand what this is for and why we need it

Reply:
Okay, great! I just got to this file, everything else looks great to me.

As for why it doesn't compile: is it this kind of an error, when trying to use it in an example, about not finding `SerdeLite`?
```
error[E0412]: cannot find type `SerdeLite` in this scope
  --> src/todo.rs:58:35
   |
58 | #[server(AddTodo, "/api", input = SerdeLite, output = SerdeLite)]
   |                                   ^^^^^^^^^ not found in this scope
   |
help: consider importing one of these items
   |
1  + use crate::server_fn::codec::SerdeLite;
   |
1  + use server_fn::codec::SerdeLite;
   |

error[E0412]: cannot find type `SerdeLite` in this scope
  --> src/todo.rs:58:55
   |
58 | #[server(AddTodo, "/api", input = SerdeLite, output = SerdeLite)]
   |                                                       ^^^^^^^^^ not found in this scope
   |
help: consider importing one of these items
   |
1  + use crate::server_fn::codec::SerdeLite;
   |
1  + use server_fn::codec::SerdeLite;
```


`codec_ident` has a magic check for idents it recognizes as being internal to the `server_fn` crate, which (unfortunately?) needs to be kept in sync with the actual set.

https://github.com/leptos-rs/leptos/blob/d8514c153eae4628eafa45511970fbde43d890bc/server_fn_macro/src/lib.rs#L834-L843

However this also reveals a bug in the parsing for `input` and `output` because they don't actually support full paths:
```rust
// should work but doesn't at present
#[server(AddTodo, "/api", input = server_fn::codec::SerdeLite, output = server_fn::codec::SerdeLite)]
```


****

Other background:

This `derives` expression describes the additional derives we need to add to the server fn type.

Recall that a
```rust
my_server_fn(foo: String, bar: u32)
```
is converted to a
```rust
struct MyServerFn {
  foo: String, 
  bar: u32
}
```
This is so that arguments can be packed into the struct, serialized, sent to the server, deserialized on the server, and unpacked from the struct, then the original fn is called with them as the arguments.

This means we need to derive some traits on the struct:
- for the `serde` encodings (`serde_json`, `ciborium` and so on), we need to add `#[derive(Serialize, Deserialize)]`
- for `rkyv`, TODO
- for `serde_lite`, we need to derive the `serde_lite` `Serialize` and `Deserialize` traits (I think?)

Using the `#server_fn_path` bit is just so that we can use the reexported `serde` or `serde_lite` dependencies.